flake.nix

```
@@ -67,6 +69,7 @@
             # Compiler toolchain
             hpkgs.ghc
             pkgs.haskellPackages.cabal-install
+            pkgs.haskellPackages.haskell-language-server
```

Not sure if changes to flake should be merged as part of this PR, but HLS has to match the GHC version

```suggestion
            hpkgs.haskell-language-server
```

---

Agreed! And whether to merge or not is a subjective decision. I put it in its own commit to easily take it out if it's not wanted. Who does have final say regarding this?